atecc608a_se.c

```diff
@@ -211,6 +211,53 @@ static psa_status_t atecc608a_import_public_key(psa_key_slot_number_t key_slot,
     return status;
 }
 
+static psa_status_t atecc608a_generate_key(psa_key_slot_number_t key_slot,
+                                           psa_key_type_t type,
+                                           psa_key_usage_t usage,
+                                           size_t bits,
+                                           const void *extra,
+                                           size_t extra_size,
+                                           uint8_t *p_pubkey_out,
+                                           size_t pubkey_out_size,
+                                           size_t *p_pubkey_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+
+    /* The hardware has slots 0-15 */
+    if (key_slot > 15)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (type != PSA_KEY_TYPE_ECC_KEYPAIR(PSA_ECC_CURVE_SECP256R1))
+    {
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (bits != PSA_BYTES_TO_BITS(ATCA_PRIV_KEY_SIZE))
```

Comment:
The core guarantees that `bits` is consistent with `type`, so this check is redundant. But it's ok to have it for robustness.

Reply:
Should this be an assert to make sure the core didn't pass us junk?

Reply:
Since I wasn't sure about the constraints that the core will have, I opted for not using asserts and just returning errors instead.